### PR TITLE
Handling of federated groups

### PR DESCRIPTION
**Pull request by aghsmith**
I've only been able to partially test the code in its current form as I solved the problem with one change and then retrospectively fixed some issues in the script. I don't have multiple test environments to test with. 

The existing code had issues if the proposed irods service group was defined externally (to the /etc/group file) e.g. via Active Directory binding.

In the setup_service_account function, 'grp.getgrall()' was used to find all groups in /etc/group and if the new service account group did not appear, it would be created with a system call to 'groupadd -r <irods_group>'. If the group in question did indeed exist, but in an external authentication system, then it would throw an exception and terminate.

I've proposed a change that wraps the group creation command in an permissive execute and then checks the return value. It will report a warning if the group exists but is non-local and continue. It will also catch other errors and throw an exception with the exception message.

This relates to a conversation on google chat:
https://groups.google.com/g/irod-chat/c/7DaQzFMLGgY

### CODE REVIEW

**Comment by trel** [comment]
ah, very good.

we'll review, and then when the time comes, please point the first commit at 6246.

and the second commit to 858 (housekeeping).

thanks!

File: scripts/setup_irods.py [hunk 1]
```diff
@@ -207,7 +207,11 @@ def setup_service_account(irods_config, irods_user, irods_group):
 
     if irods_group not in [g.gr_name for g in grp.getgrall()]:
         l.info('Creating Service Group: %s', irods_group)
-        irods.lib.execute_command(['groupadd', '-r', irods_group])
+        createGroup = irods.lib.execute_command_permissive(['groupadd', '-r', irods_group])
```
[the remark this answers is not in the document]
Comment (aghsmith) [reply scripts/setup_irods.py]: Thanks Alan, I've implemented this.

**Comment by aghsmith** [comment]
@trel I've added the issue numbers to the commits as requested.

**Review by korydraughn — APPROVED** [review]

**Comment by alanking** [comment]
Excellent. Merging on green.

**Review by trel — APPROVED** [review]

**Pull request merged**